src/week08/lecture/staking.py

```diff
@@ -0,0 +1,43 @@
+from opshin.prelude import *
+
+
+def equal(a: Union[StakingHash, StakingPtr], b: Union[StakingHash, StakingPtr]) -> bool:
+    result = False
+    if isinstance(b, StakingHash):
+        if isinstance(a, StakingHash):
+            result = b == a
+    elif isinstance(b, StakingPtr):
+        if isinstance(a, StakingPtr):
+            result = b == a
+    return result
+
+
+def get_amount(context: ScriptContext, purpose: Rewarding) -> int:
+    amount = 0
+    found = False
+    for staking_cred, a in context.tx_info.wdrl.items():
+        if equal(staking_cred, purpose.staking_credential):
+            found = True
+            amount = a
+    assert found, "withdrawal not found"
+    return amount
+
+
+def paid_to_address(addr: Address, context: ScriptContext) -> int:
```

Comment:
Can you re-use `all_tokens_locked_at_address` for this? The token should be Lovelace in this case

https://github.com/OpShin/opshin/blob/c252101628cc318fdef1616105698750a438885b/opshin/prelude.py#L47